open-metadata-implementation/adapters/open-connectors/event-bus-connectors/open-metadata-topic-connectors/kafka-open-metadata-topic-connector/src/main/java/org/odpi/openmetadata/adapters/eventbus/topic/kafka/KafkaOpenMetadataEventProducer.java

```diff
@@ -119,7 +124,11 @@ private void publishEvent(String event) throws ConnectorCheckedException
                 if( !isExceptionRetryable(error)) {
                     /* kafka thinks this isn't a retryable problem */
                     /* so let the caller try */
-                    log.error("Exception in sendEvent " + error.toString());
+
+                    producer.close();
+                    producer = null;
+
+                    log.error("Execution Exception closed producer ");
```

Comment:
I don't think we need to log an error here given that we're throwing an exception in the lines immediately following?